src/object/cli_obj.c

```diff
@@ -4386,6 +4389,9 @@ obj_comp_cb(tse_task_t *task, void *data)
 		     task->dt_result == -DER_CSUM)))
 			obj_auxi->io_retry = 0;
 
+		if (task->dt_result == -DER_SHARDS_OVERLAP)
```

Comment:
Under which cases the DER_SHARDS_OVERLAP will be returned via obj_comp_cb()? As my understand, during obj_req_get_tgts(), we will check whether there are two shares locate on the same target or not. If we pass such check, then where the -DER_SHARDS_OVERLAP will returned later?

Reply:
yes, obj_req_get_tgts() will return OVERLAP if there are two shards on the same target, then obj_comp_cb() will be called and tx_convert is bet set there, which will trigger retry, then go into dc_tx_convert.

In original implementation, it did sth like

if (error) {
             obj_comp_cb();
              tse_task_complete();
}

Even obj_comp_cb() may see the task needs to retry, and re-init the task, but tse_task_complete() may still terminate the task.